src/linker/Linker/XmlDependencyRecorder.cs

```diff
@@ -89,6 +89,12 @@ public void Dispose ()
 			stream = null;
 		}
 
+		public void RecordDependency (object target, DependencyInfo reason, bool marked)
+		{
+			// Used for the DependencyInfo-based recorders. Don't do anything as we only
```

Comment:
Can we simply call `RecordDependency(reason.Source, target, marked)` (and eventually adapt this to the new tracing and actually preserve the reasons as well).

Reply:
I wanted to avoid producing a lot of redundant output in the xml (which this would do since the stack-based tracer has a fair amount of overlap with the tracing that I'm adding). Or do you mean to remove the stack-based tracing?

Reply:
I thought that the only reason we needed the "stack" in the existing tracer is the fact that we don't report source->target pairs with each "Mark" action. Now that we do, there should be no need for the stack.

Reply:
I went ahead and removed the stack-based tracer.